packages/core/src/pubsub/message-bus.ts

```diff
@@ -0,0 +1,69 @@
+import { buildQueryMessage, MsgType, PubsubMessage, ResponseMessage } from './pubsub-message';
+import { Observable, Subject, Subscription, SubscriptionLike } from 'rxjs';
+import { filter, map, take, tap } from 'rxjs/operators';
+import { DocID } from '@ceramicnetwork/docid';
+import CID from 'cids';
+
+/**
+ * Multiplexing IPFS Pubsub.
+ */
+export class MessageBus extends Observable<PubsubMessage> implements SubscriptionLike {
+  private readonly pubsubSubscription: Subscription;
+  private readonly feed$: Subject<PubsubMessage> = new Subject<PubsubMessage>();
+
+  constructor(
+    readonly pubsub: Observable<PubsubMessage> & { next: (m: PubsubMessage) => Subscription },
```

Comment:
woah, what does this syntax do?

Reply:
It states `pubsub` is an Observable _and_ has `next` method. I intend to give a `SubjectLike` name to this interface at the next iteration.